### ISSUE TITLE
Question about Logging in JHOVE

### ISSUE THREAD

**Issue by ross-spencer** [openpreserve/jhove#228]
I noticed today that there is very little by way of logging in JHOVE if i search for logger.log

https://github.com/openpreserve/jhove/search?utf8=%E2%9C%93&q=logger.log&type=

I  was interested to generate more messages to help me understand what processing was happening while working through a few bugs. 

Is there a way to generate more output from the tool, or is it something that should be considered in future enhancements?

**Comment by carlwilson** [comment]
Hi @ross-spencer am looking at this now, will write up a brief few notes after I test what I'm about to say is true...

**Comment by carlwilson** [comment]
OK, this was interesting, I found a problem, as it stands the logic prevents the JHOVE CLI app from ever honouring the logging configuration. I've fixed raised this as issue #237 and fixed it with a soon to be merged PR #238

With that out of the @ross-spencer here's a VERY rough guide to logging. Feel free to add loggers elsewhere, but do use native Java logging rather than introduce a dependency. The correct import and logger initialization look like:

```java
import java.util.logging.Level;
import java.util.logging.Logger;

public class SomeClass {
    /** Logger for this class. */
    private static final Logger logger = Logger.getLogger(SomeClass.class.getCanonicalName());
}
```
The logger honours the concept of logging from `Level.FINEST` (detailed debug) to `Level.SEVERE` for unrecoverable errors. For the message levels you're talking about any of the `FINE, FINER, or FINEST` settings would work. You can then log at your chosen level using:
```java
logger.log(Level.FINE, "meesage")
logger.log(Level.FINE, "meesage", exception)
```

To change the logging level when you run the code, you can either add something like this to the config file:

```xml
<jhoveConfig version="1.0"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 xmlns="http://hul.harvard.edu/ois/xml/ns/jhove/jhoveConfig"
 xsi:schemaLocation="http://hul.harvard.edu/ois/xml/ns/jhove/jhoveConfig
                     http://hul.harvard.edu/ois/xml/xsd/jhove/1.6/jhoveConfig.xsd">
 <jhoveHome>/home/cfw/jhove-dev</jhoveHome>
 <defaultEncoding>utf-8</defaultEncoding>
 <bufferSize>131072</bufferSize>
 <module>
  ...
 </module>
  <logLevel>ALL</logLevel>
</jhoveConfig>
```
Where the log level honours the following values:

```xml
<xs:enumeration value="OFF"/>
<xs:enumeration value="CONFIG"/>
<xs:enumeration value="INFO"/>
<xs:enumeration value="WARNING"/>
<xs:enumeration value="SEVERE"/>
<xs:enumeration value="FINE"/>
<xs:enumeration value="FINER"/>
<xs:enumeration value="FINEST"/>
<xs:enumeration value="ALL"/>
```
So setting `<logLevel>ALL</logLevel>` shows all message, where `<logLevel>WARNING</logLevel>` would display `WARNING` and `SEVERE` log messages only. JHOVE's default settings show only `SEVERE` messages.

You can also change the logging level by passing a `-l` param with the same level values, e.g. `jhove --version -l ALL` would set the log level to `ALL` and overrides the config setting.

**Comment by ross-spencer** [comment]
This is great Carl. Maybe one for the next DIG call as well, or the next hack day, but we should translate this to a documentation/guide somewhere. Once the release is done we can create example screenshots for logging as well if it will be useful for the community.
